Home_Work_2.py

```diff
@@ -0,0 +1,178 @@
+from pydantic import BaseModel
+from pprint import pprint
+from abc import ABC, abstractmethod
+from functools import wraps
+
+
+class BookModel(BaseModel):
+    name: str
+    author: str
+    issue_date: str
+
+
+class AbstractClass(ABC):
+    @abstractmethod
+    def __init__(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def __str__(self):
+        raise NotImplementedError
+
+
+class Book(AbstractClass):
+    name = None
+    __author = None
+    issue_date = None
```

Comment:
It's not mandatory to define these attributes here, because you alreade have them in `__init__` method